### PR TITLE
connect the api with the client app

### PR DESCRIPTION
**Pull request by VladBejenariu**
connect the api with the client to create and confirm a setup intent operation

### CODE REVIEW

File: payment-api/PaymentApi/Services/StripeClient.cs [hunk 1]
```diff
@@ -40,7 +40,7 @@ public async Task<string> CreateCustomerAsync(string firstName, string lastName,
                 Description = $"{firstName} {lastName}",
                 Email = email,
                 Name = $"{firstName} {lastName}",
-                TaxIdData = new List<CustomerTaxIdDataOptions> { }
+                // TaxIdData = new List<CustomerTaxIdDataOptions> { }
```
Comment (Sebyd) [inline payment-api/PaymentApi/Services/StripeClient.cs:43]: Does this crash?

File: payment-api/PaymentApi/Features/CreatePaymentSetupIntent.cs [hunk 2]
```diff
@@ -23,20 +23,20 @@ public class CommandValidator : AbstractValidator<Command>
         {
             public CommandValidator()
             {
-                RuleFor(x => x.UserId)
-                    .NotEmpty();
+                //RuleFor(x => x.UserId)
```
Comment (Sebyd) [inline payment-api/PaymentApi/Features/CreatePaymentSetupIntent.cs:26]: Cann we pass them as parameters in request instead?

File: payment-api/PaymentApi/Services/StripeClient.cs (hunk 1 above)
Comment (VladBejenariu) [reply payment-api/PaymentApi/Services/StripeClient.cs:43]: Yes. Message from logs: "You passed an empty string for 'tax_id_data'. We assume empty values are an attempt to unset a parameter; however 'tax_id_data' cannot be unset. You should remove 'tax_id_data' from your request or supply a non-empty value."

File: payment-api/PaymentApi/Features/CreatePaymentSetupIntent.cs (hunk 2 above)
Comment (VladBejenariu) [reply payment-api/PaymentApi/Features/CreatePaymentSetupIntent.cs:26]: I do not get how these are used atm, for example the /payment-intent post does not have this body, so I am not sure where to set set them

Comment (Sebyd) [reply payment-api/PaymentApi/Features/CreatePaymentSetupIntent.cs:26]: Ok. I will update the request method parameters later.

**Review by Sebyd — APPROVED** [review]

**Pull request merged**